.buildkite/pipeline.yml

```diff
@@ -12,27 +12,48 @@
 # permissions and limitations under the License.
 
 steps:
-  - label: ':go: go mod download'
-    command: 'go mod download'
 
-  # We use a "wait" step here, because Go's module logic freaks out when
-  # multiple go builds are downloading to the same cache.
-  - wait
+  - label: ":docker: Build"
+    env:
+      DOCKER_BUILDKIT: "1"
+    command:
+      >
+        docker build \
+          --progress=plain \
+          -f tools/docker/Dockerfile \
+          --target firecracker-containerd-test-e2e \
+          -t firecracker-containerd-test-e2e .
```

Comment:
We should use something like `--tag "localhost/firecracker-containerd-test-e2e:${BUILDKITE_BUILD_NUMBER}"` so that parallel invocations of `docker build` don't stomp over each other.